ecosystem/indexer-grpc/indexer-grpc-utils/src/cache_operator.rs

```diff
@@ -260,13 +264,34 @@ impl<T: redis::aio::ConnectionLike + Send + Clone> CacheOperator<T> {
         let cache_coverage_status = self.check_cache_coverage_status(start_version).await;
         match cache_coverage_status {
             Ok(CacheCoverageStatus::CacheHit(v)) => {
+                if let Some(cache) = self.read_through_cache.clone() {
+                    let keys = start_version..start_version + v;
+                    let mut results = vec![];
+                    for key in keys {
+                        if let Some(value) = cache.get(&key) {
+                            results.push(value);
+                        } else {
+                            break;
+                        }
+                    }
+                    if results.len() == v as usize {
+                        return Ok(CacheBatchGetStatus::Ok(results));
```

Comment:
this uses the cache results if all the items are in the passthrough cache. can it happen that it gets some txn's from cache and some txn's from redis and combine/sort the result?

Reply:
there are multiple improvements we can do as we discussed
1. (highly preferred) update lock --- i don't have time to finish..
2. partial update as you mentioned - i'll implement this and land on to devnet. 

more details in https://github.com/aptos-labs/aptos-core/pull/11992